pkg/cmd/status/statusBuilder.go

```diff
@@ -45,46 +50,56 @@ type registryStatus struct {
 }
 
 type clientConfig struct {
-	config     config.IConfig
-	Logger     logging.Logger
-	connection connection.Connection
-	localizer  localize.Localizer
+	context       context.Context
+	config        config.IConfig
+	Logger        logging.Logger
+	connection    connection.Connection
+	localizer     localize.Localizer
+	serviceConfig *servicecontext.ServiceConfig
 }
 
 type statusClient struct {
-	config    config.IConfig
-	Logger    logging.Logger
-	conn      connection.Connection
-	localizer localize.Localizer
+	context       context.Context
+	config        config.IConfig
+	Logger        logging.Logger
+	conn          connection.Connection
+	localizer     localize.Localizer
+	serviceConfig *servicecontext.ServiceConfig
 }
 
 // newStatusClient returns a new client to fetch service statuses
 // and build it into a service status config object
 func newStatusClient(cfg *clientConfig) *statusClient {
 	return &statusClient{
-		config:    cfg.config,
-		Logger:    cfg.Logger,
-		conn:      cfg.connection,
-		localizer: cfg.localizer,
+		config:        cfg.config,
+		Logger:        cfg.Logger,
+		conn:          cfg.connection,
+		localizer:     cfg.localizer,
+		serviceConfig: cfg.serviceConfig,
 	}
 }
 
-// BuildStatus gets the status of all services currently set in the user config
-func (c *statusClient) BuildStatus(ctx context.Context, services []string) (status *serviceStatus, ok bool, err error) {
-	cfg, err := c.config.Load()
-	if err != nil {
-		return nil, false, err
-	}
+// BuildStatus gets the status of all services currently set in the service context
+func (c *statusClient) BuildStatus(ctxName string, services []string) (status *serviceStatus, ok bool, err error) {
 
 	status = &serviceStatus{}
 
+	status.Name = ctxName
+
+	if rhoasContext := os.Getenv("RHOAS_CONTEXT"); rhoasContext != "" {
+		status.Location = rhoasContext
+	} else {
+		ctxDirLocation, _ := servicecontext.DefaultDir()
+		status.Location = filepath.Join(ctxDirLocation, "contexts.json")
+	}
+
 	if stringInSlice(servicespec.KafkaServiceName, services) {
-		if instanceID, exists := cfg.GetKafkaIdOk(); exists {
+		if c.serviceConfig.KafkaID != "" {
 			// nolint:govet
-			kafkaStatus, err := c.getKafkaStatus(ctx, instanceID)
+			kafkaStatus, err := c.getKafkaStatus(c.context, c.serviceConfig.KafkaID)
```

Comment:
I think we should use context helper method to get kafka request instead and pass it to getKafkaStatus.
Same for the service registry. Because this implementation is custom we are missing validation right now.

That will fix problem when we use fake id:
[wtrocki@graphapi app-services-cli (context-cmd ✗)]$ rhoas context status
❌ 404 Not Found. Run the command in verbose mode using the -v flag to see more information